os module, and executeCommand function

miniche opened the pull request:
This function will be use everywhere we need to execute an os command.
It's more flexible, if we want to replace the "subprocess.Popen" call in the future for example. Returns the result string.

Todo: 
- [x] Create the os module
- [x] Design the executeCommand function
- [x] Replace code everywhere is needed to use the new function instead of "subprocess" call

miniche committed:
Added os module, with "executeCommand" function.

This function will be use everywhere we need to execute an os command.
It's more flexible, if we want to replace the "subprocess.Popen" call in the future for example.
Returns the result string.

miniche committed:
Improved os.executeCommand function.

executeCommand function catch now correctly pipes and vars with double quotes.
We use "shlex.split" in POSIX mode, this python function returns the command formatted for subprocess module.

Ninir commented on centralreport/cr/os.py:
```diff
@@ -0,0 +1,44 @@
+
+def WIP_executeCommand(str_command):
+    """
+        This function executes an os command. Command can contains pipes, executed as subcommands.
```
``` python
str_command can contain pipes
```

'Find it weird to have "Command", which is implicitly related to the str_command parameter

Ninir commented on centralreport/cr/os.py:
```diff
@@ -0,0 +1,44 @@
+    """
+        This function executes an os command. Command can contains pipes, executed as subcommands.
+        If your command contains vars with a space, use " " to escape words (eg: "Volume Name")
+        If you want to use an double quote (") in your command, escape it with \\ (eg: \\":\\" return ":")
```
Typo: `an`

Ninir commented on centralreport/cr/os.py:
```diff
@@ -0,0 +1,44 @@
+        Returns the result of the command (string).
+    """
+
+    # Command can contains pipes. At each pipe, we execute a "subcommand".
```
Typo: s to contain

Ninir commented on centralreport/cr/os.py:
```diff
@@ -0,0 +1,44 @@
+            list_results.append(subprocess
+                                .Popen(list_command, stdout=subprocess.PIPE, stdin=list_results[i - 1]
+                                .stdout))
+            # End the previous subprocess (http://docs.python.org/2/library/subprocess.html#replacing-shell-pipeline)
```
Typo: Ends

miniche committed:
Spelling fixes in os module

miniche committed:
Renamed "os" module to "system"

Big mistake: "os" is already a python core module. We cannot use this name, or our scripts will not able to find the native module.

miniche committed:
In collectors, use new executeCommand function instead of subprocess calls

It's more readable and convenient, no? :)

miniche commented:
Time to review! :innocent:

Guimove commented:
Seems to be good for me
